Purchases/Attribution/AttributionFetcher.swift

```diff
@@ -43,16 +43,7 @@ class AttributionFetcher {
     }
 
     var identifierForVendor: String? {
-        // Should match available platforms in
-        // https://developer.apple.com/documentation/uikit/uidevice?language=swift
-        // https://developer.apple.com/documentation/watchkit/wkinterfacedevice?language=swift
-#if os(iOS) || os(tvOS)
-        UIDevice.current.identifierForVendor?.uuidString
-#elseif os(watchOS)
-        WKInterfaceDevice.current().identifierForVendor?.uuidString
-#else
-        nil
-#endif
+        return self.systemInfo.identifierForVendor
```

Comment:
🤦‍♂️ and the implementations are slightly different.

Reply:
Were they supposed to be? Or are you commenting on the fact that they got out of sync?

Reply:
Sorry, I should have been clearer, I was commenting on the fact they got out of sync.

Reply:
Oh ok :P I was worried for a second that there was a reason they were different 😝

Reply:
haha no, we're good. They got out of sync recently #991. 
For context, the reason we had this twice was also pre-migration stuff - we had it originally in objc, needed it from swift but couldn't move it because of the dependencies of the class, so we did it again in swift, and then forgot to unify during the migration.

Reply:
Hehe makes sense.